src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -145,10 +146,15 @@ public override int Run(InterpretedFrame frame)
 
         private sealed class GetValueOrDefault : NullableMethodCallInstruction
         {
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
             private readonly Type _defaultValueType;
 
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2074:UnrecognizedReflectionPattern",
```

Comment:
This suppression will only be correct if we replace the Activator.CreateInstance with GetUninitializedObject. Are we sure we won't forget to revisit this if the resolution of the new bug is unexpected?

Reply:
Are you saying it isn't correct because trimming will cause `Activator.CreateInstance` to fail? Or are you saying it is incorrect based on if value types get a parameterless constructor, and the trimmer removes it, causing a behavior change between trimmed and untrimmed apps?

Either way I've called this out in #47647.

Reply:
The latter. The annotation on the field is writing checks no consumer can cash (because the suppression that makes is possible is incorrect).

I would keep the suppression on the `Run` method where it was before because once we swap this for GetUnitializedObject we can say something along the lines of "this is okay to suppress because trimming always considers valuetypes boxed" - which is what we need for the suppression on `GetUninitializedObject` to be safe.

Reply:
In the spirit of this comment (and remembering I made the same suppression elsewhere), I tried changing the following code from `Activator.CreateInstance` to `RuntimeHelpers.GetUninitializedObject`:

https://github.com/dotnet/runtime/blob/d1c0fa82435ec5774f038dd7f33dfada1f1083b9/src/libraries/Common/src/Extensions/ParameterDefaultValue/ParameterDefaultValue.cs#L42-L50

However, unit tests started failing. This is because `RuntimeHelpers.GetUninitializedObject` using a `Nullable<T>` type doesn't return `null`, like `Activator.CreateInstance`:

```C#
t = typeof(Nullable<int>);
Console.WriteLine($"Activator Is Null? {Activator.CreateInstance(t) == null}");
Console.WriteLine($"GetUninitializedObject Is Null? {RuntimeHelpers.GetUninitializedObject(t) == null}");
```

prints

```
Activator Is Null? True
GetUninitializedObject Is Null? False
```

This won't be a problem for this specific site, since you can't have a `Nullable<Nullable<T>>`. But for ParameterDefaultVale (and potentially other places), these two APIs aren't interchangeable.

Reply:
Good catch. Something we need to keep in mind as we go fixing incorrect usages of CreateInstance.